File: Crio/Array/4_ShortestUnsortedContinuousSubarray/AlternateApproach.py
# Implement your solution here
def find_unsorted_subarray(nums):
    if len(nums) <= 1 :
        return 0
    sortedNums = nums[:]
    sortedNums.sort()
    l,r,n = 0,0,len(nums)
    for l in range(len(nums)) :
        if(nums[l]!=sortedNums[l]) :
            break
    for r in range(len(nums)) :
        index = n-1-r
        if(nums[index]!=sortedNums[index]) :
            break
    r = n-r-1
    ans = max(r - l + 1,0)
    return ans

def main():
    n = int(input())
    nums = list(map(int, input().split()))
    answer = find_unsorted_subarray(nums)
    print(answer)


if __name__ == "__main__":
    main()

File: Crio/Array/4_ShortestUnsortedContinuousSubarray/AlternateApproach.cpp
#include<bits/stdc++.h>
using namespace std;

int findUnsortedSubarray(vector<int>& nums) { 
	int i,j,n=nums.size();
	vector<int> nums_copy;
	
	for(i=0;i<n;i++) {
		nums_copy.push_back(nums[i]);
	}
	sort(nums_copy.begin(),nums_copy.end());
	for(i=0;i<n;i++) {
		if(nums_copy[i]!=nums[i])
			break;
	}
	for(j=n-1;j>=0;j--) {
		if(nums_copy[j]!=nums[j])
			break;
	}

	if(i>j)
	  return 0;
	else
	  return j-i+1;
}

int main()
{
	int n;
	cin >> n;
	vector<int>nums(n);
	for(int i = 0 ; i < n ; i++) {
		cin >> nums[i];
	}
	int answer = findUnsortedSubarray(nums);
	cout << answer << "\n";
	return 0;	
}
